serial/pet/driver.s

```diff
@@ -0,0 +1,730 @@
+
+BAUD            = 4                   ; Baud rate, index into BaudTblLo/Hi (4 = 2400)
+
+;-----------------------------------------------------------------------------------
+; Constants
+;-----------------------------------------------------------------------------------
+
+ST_STARTBIT     = 0                   ; Waiting/Sending for start bit
+ST_READY        = 1                   ; Ready to start sending
+ST_DATABIT      = 2                   ; Sending/receiving data
+ST_STOPBIT      = 3                   ; Sending/receiving stop bit
+ST_IDLE         = 4
+
+BITCOUNT        = 8                   ; 8-bit bytes to recieve
+
+; 6520 PIAs - Not in CC65 pet.inc
+PIA1_PA         = $E810
+PIA1_PB         = $E812
+PIA1_CRA        = $E811
+PIA1_CRB        = $E813
+
+PIA2_PB         = $E822
+PIA2_CRA        = $E821
+PIA2_CRB        = $E823
+
+BASIC_VECT_IRQ  = $0090               ; 90/91 - Hardware interrupt vector
+
+;-----------------------------------------------------------------------------------
+; OpenSerial: Setup the code to handle RS232 I/O
+;-----------------------------------------------------------------------------------
+
+OpenSerial:
+                sei                   ; Disable interrupts
+
+                ; Disable all PIA interrupt sources
+                lda PIA1_CRB
+                and #$FE              ; Disable interrupts (60hz retrace interval?)
+                sta PIA1_CRB
+                lda PIA1_CRA
+                and #$FE
+                sta PIA1_CRA          ; Disable interrupts
+
+                lda PIA2_CRB
+                and #$FE              ; Disable interrupts (60hz retrace interval?)
+                sta PIA2_CRB
+                lda PIA2_CRA
+                and #$FE
+                sta PIA2_CRA          ; Disable interrupts
+
+                lda BASIC_VECT_IRQ
+                sta IrqBasicLo        ; Save IRQ lo byte for BASIC
+                lda BASIC_VECT_IRQ+1
+                sta IrqBasicHi        ; Save IRQ hi byte for BASIC
+
+                ; Install IRQ
+                lda #<IrqHandler
+                ldx #>IrqHandler
+                sta BASIC_VECT_IRQ
+                stx BASIC_VECT_IRQ+1
+
+                lda #$00              ; Rx pin in (PA0) (rest input as well)
+                sta VIA_DDRA          ; Set directions
+                lda #$40              ; Shift register disabled, no latching, T1 free-run, T2 one-shot
+                sta VIA_CR
+
+                lda #$EC              ; Tx as output high, uppercase+graphics ($EE for lower)
+                                      ; CA1 trigger on falling edge
+                sta VIA_PCR
+                ; Set VIA interrupts so that our timer is the only interrupt source
+                lda #$7F              ; Clear all interrupt flags
+                sta VIA_IER
+                lda #$C2              ; Enable Timer 1 interrupt and CA1 interrupt
+                sta VIA_IER
```

Comment:
Just want to make sure this is what is intended.  I assume it latches it or something but its two stores to the same location./

Reply:
The first write explicitly disables all interrupts (bit 7 in the value written is 0), the second then explicitly enables two interrupts (bit 7 written is 1). The only way to make sure you achieve the desired result of having only two interrupts enabled is two consecutive writes.